src/SoftRobots/component/constraint/model/CableModel.cpp

```diff
@@ -41,12 +40,158 @@ namespace component
 namespace constraintset
 {
 
-using namespace sofa::defaulttype;
+
+using sofa::defaulttype::Vec2Types;
+using sofa::type::Vector2;
+
+template<>
+unsigned int CableModel<Vec2Types>::computeClosestIndice(Coord position)
+{
+    double closest_distance = (position - Coord(m_topology->getPX(0), m_topology->getPY(0))).norm();
```

Comment:
When dealing with the floating point types, use the one provided by the `DataTypes`. Here, you can use `Vec2Types::Real`.
Therefore, `double closest_distance = ...` would become `Vec2Types::Realclosest_distance = ...`.
Or you can let the compiler handling it automatically:
`auto closest_distance = (position - Coord(m_topology->getPX(0), m_topology->getPY(0))).norm();`

This suggestion applies for the rest of the code.